scipy/spatial/transform/_rotation.pyx

```diff
@@ -223,6 +230,131 @@ cdef double[:, :] _compute_euler_from_matrix(
 
     return angles
 
+@cython.boundscheck(False)
+@cython.wraparound(False)
+cdef double[:, :] _compute_euler_from_quat(
+    np.ndarray[double, ndim=2] quat, const uchar[:] seq, bint extrinsic=False
+):
+    # The algorithm assumes intrinsic frame transformations. The algorithm
+    # in the paper is formulated for rotation matrices which are transposition
+    # rotation matrices used within Rotation.
+    # Adapt the algorithm for our case by
+    # 1. Instead of transposing our representation, use the transpose of the
+    #    O matrix as defined in the paper, and be careful to swap indices
+    # 2. Reversing both axis sequence and angles for extrinsic rotations
+    
+    if not extrinsic:
+        seq = seq[::-1]
+
+    cdef int i = _elementary_basis_index(seq[0])
+    cdef int j = _elementary_basis_index(seq[1])
+    cdef int k = _elementary_basis_index(seq[2])
+
+    is_proper = i == k
+    if is_proper:
+        k = 3 - i - j # get third axis
+        
+    # Step 0
+    # Check if permutation is even (+1) or odd (-1)     
+    cdef int sign = int((i-j)*(j-k)*(k-i)/2)
+
+    cdef Py_ssize_t num_rotations = quat.shape[0]
+
+    # some forward definitions
+    cdef double[:, :] angles = _empty2(num_rotations, 3)
+    cdef double[:] _angles # accessor for each rotation
+    cdef double a, b, c, d, n2
+    cdef double half_sum, half_diff
+    cdef double eps = 1e-7
+    cdef bint safe1, safe2, safe, adjust
+
+    for ind in range(num_rotations):
+        _angles = angles[ind, :]
+
+        # Step 1
+        # Permutate quaternion elements            
+        if is_proper:
+            a = quat[ind, 3]
+            b = quat[ind, i]
+            c = quat[ind, j]
+            d = quat[ind, k] * sign
+        else:
+            a = quat[ind, 3] - quat[ind, j]
+            b = quat[ind, i] + quat[ind, k] * sign
+            c = quat[ind, j] + quat[ind, 3]
+            d = quat[ind, k] * sign - quat[ind, i]
+    
+        # Step 2
+        # Compute second angle...
+        n2 = a**2 + b**2 + c**2 + d**2
```

Comment:
Now I've noticed. This combination should be always 1 as we use normalized quaternions internally. Am I right?

Reply:
In theory, yes!
 
In practice, since I'm doing this calculation here, I left out the division by `sqrt(2)` needed in the four lines where I define `a`,`b`,`c` and `d` in case of Tait Bryan angles.

If you want to remove the calculation of `n2` line and the division by it, then this:
```
a = quat[ind, 3] - quat[ind, j]
b = quat[ind, i] + quat[ind, k] * sign
c = quat[ind, j] + quat[ind, 3]
d = quat[ind, k] * sign - quat[ind, i]
```
Should be changed to this:
```
a = (quat[ind, 3] - quat[ind, j])/sqrt(2)
b = (quat[ind, i] + quat[ind, k] * sign)/sqrt(2)
c = (quat[ind, j] + quat[ind, 3])/sqrt(2)
d = (quat[ind, k] * sign - quat[ind, i])/sqrt(2)
```